tests/testsuite/inheritable_workspace_fields.rs

```diff
@@ -1356,3 +1356,145 @@ Caused by:
         )
         .run();
 }
+
+#[cargo_test]
+fn warn_inherit_def_feat_true_member_def_feat_false() {
+    Package::new("dep", "0.1.0")
+        .feature("default", &["fancy_dep"])
+        .add_dep(Dependency::new("fancy_dep", "0.2").optional(true))
+        .file("src/lib.rs", "")
+        .publish();
+
+    Package::new("fancy_dep", "0.2.4").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "bar"
+            version = "0.2.0"
+            authors = []
+            [dependencies]
+            dep = { workspace = true, default-features = false }
+
+            [workspace]
+            members = []
+            [workspace.dependencies]
+            dep = { version = "0.1.0", default-features = true }
+        "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("build")
```

Comment:
Can you switch to "check" in tests like this? `build` is a little slower and uses more disk space.